test/api-helper/project/cachingProjectLoader.test.ts

```diff
@@ -30,8 +33,67 @@ describe("cachingProjectLoader", () => {
         const p = InMemoryProject.from(id);
         const pl = new SingleProjectLoader(p);
         const cp = new CachingProjectLoader(pl);
-        const p1 = await save(cp, {id, credentials: undefined, readOnly: true});
+        const p1 = await save(cp, { id, credentials: undefined, readOnly: true });
         assert(p1 as any === p);
     });
 
+    describe("should respect ProjectLoadingParameters", () => {
+
+        let clonedCalledTimes: number;
+        let originalCloned: any;
+        let sha: string;
+        let tmpBaseDir: any;
+
+        before(() => {
+            originalCloned = Object.getOwnPropertyDescriptor(GitCommandGitProject, "cloned");
+        });
+
+        beforeEach(() => {
+            clonedCalledTimes = 0;
+            sha = computeShaOf(Math.random().toString(36).substring(2));
+            tmpBaseDir = dirSync();
+
+            Object.defineProperty(GitCommandGitProject, "cloned", {
+                value: () => {
+                    clonedCalledTimes++;
+                    return {
+                        baseDir: tmpBaseDir.name,
+                        id: {
+                            sha: `${sha}`,
+                        },
+                    };
+                },
+            });
+        });
+
+        afterEach(() => {
+            tmpBaseDir.removeCallback();
+        });
+
+        after(() => {
+            Object.defineProperty(GitCommandGitProject, "cloned", originalCloned);
+        });
+
+        it("disparate clone params", async () => {
+
+            const id = new GitHubRepoRef("a", "b", sha);
+            const cp = new CachingProjectLoader();
+            await cp.doWithProject({ id, credentials: undefined, readOnly: true, cloneOptions: { depth: 1 } }, async () => { });
+            await cp.doWithProject({ id, credentials: undefined, readOnly: true, cloneOptions: { depth: 5 } }, async () => { });
+            await cp.doWithProject({ id, credentials: undefined, readOnly: true, cloneOptions: { depth: 5, alwaysDeep: false } }, async () => { });
```

Comment:
Per my comment above, the last two invocations are asking for the equivalent clone.